### DISCUSSION TITLE
to_param with ".html" fails to edit

### DISCUSSION THREAD

**Discussion by Olli** [avo-hq/avo#3451] [Help] [answered]
Hi,
in my models I have a to_param which builds me something like "#{urlalias}.html" (for legacy reasons). If I want to edit this resource I got an error that the route can't be found. 
Is there any way to handle this problem?

**Comment by Paul-Bob** [comment]
Hi @Olli

Unfortunately we didn't found a solution for this issue yet

There is a opened issue around this, let's continue the discussion there https://github.com/avo-hq/avo/issues/3368

**Comment by Olli** [comment, marked as answer]
Well maybe "get "/avo/resources/:resource/:id.html/edit", to: redirect("/avo/resources/%{resource}/%{id}/edit")" should resolve this issue at least on my side. Unfortunately it took some hours for me to find this solution thou it's so easy.

**Reply by Paul-Bob** [reply]
Hmm interesting, can you share a snippet with an example please?

**Reply by Olli** [reply]
Im not entirely sure what snippet you like but here is my routes.rb configuration
```
  authenticate :user do
    get "/avo/resources/:resource/:id.html/edit", to: redirect("/avo/resources/%{resource}/%{id}/edit")
    mount Avo::Engine, at: Avo.configuration.root_path
  end
```
in my model I have
```
to_param
  "#{urlalias}.html"
end
```
I should mention, that the redirect doesn't work if the urlalias contains a German "ß". I'm not sure why but tbh I don't understand the Rails routing code. In theory "ß" shouldn't be a problem. 
If the "ß" is in the url alias the ":id" can't be recognized and the route can't be found.

**Reply by Olli** [reply]
so /avo/resources/my-resource/resourcealias.html/edit is redirected to /avo/resources/my-resource/resourcealias/edit

**Reply by Paul-Bob** [reply]
Thanks for sharing this @Olli!!
